lib/model/migrations/20231002-01-add-conflict-details.js

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = async (db) => {
+  await db.raw(`CREATE TYPE "conflictType" AS ENUM ('soft', 'hard')`); // eslint-disable-line quotes
+
+  await db.raw('ALTER TABLE entities ADD COLUMN conflict "conflictType" NULL');
+
+  await db.raw(`ALTER TABLE entity_defs 
+    ADD COLUMN "dataReceived" JSONB NOT NULL,
```

Comment:
When I tried to run the migration, I got this error
```
column "dataReceived" of relation "entity_defs" contains null values
```

I think this column should have a default like ` DEFAULT '{}'::jsonb`